test/jdk/javax/swing/plaf/gtk/JTableCtrlShiftRightLeftKeyTest.java

```diff
@@ -0,0 +1,139 @@
+    private static Robot robot;
+    private static final int SELECTED_COLUMN = 2;
+
+    public static void main(String[] args) throws Exception {
+        UIManager.setLookAndFeel("com.sun.java.swing.plaf.gtk.GTKLookAndFeel");
+        robot = new Robot();
+        robot.setAutoDelay(50);
+        try {
+            SwingUtilities.invokeAndWait(JTableCtrlShiftRightLeftKeyTest::createAndShowUI);
+            robot.waitForIdle();
+            robot.delay(1000);
+
+            SwingUtilities.invokeAndWait(() -> {
+                tableLoc = table.getLocationOnScreen();
+                cellRect = table.getCellRect(0, SELECTED_COLUMN, true);
+            });
+
+            robot.mouseMove(tableLoc.x + cellRect.x + cellRect.width / 2,
+                    tableLoc.y + cellRect.y + cellRect.height / 2);
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+            robot.waitForIdle();
+            robot.delay(100);
+
+            testCtrlShift(KeyEvent.VK_RIGHT, SELECTED_COLUMN,
+                    table.getColumnCount() - 1, "RIGHT");
+
+            robot.waitForIdle();
+            robot.delay(100);
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+            robot.waitForIdle();
+            robot.delay(100);
+
+            testCtrlShift(KeyEvent.VK_LEFT, 0,
+                    SELECTED_COLUMN, "LEFT");
+            robot.waitForIdle();
+            robot.delay(100);
+            System.out.println("Test Passed!");
+
+        } finally {
+            SwingUtilities.invokeAndWait(() -> {
+                if (frame != null) {
+                    frame.dispose();
+                }
+            });
+        }
+    }
+
+    private static void testCtrlShift(int keySelected, int startCellCheck,
+                                      int endCellCheck, String key) throws Exception {
+        robot.keyPress(KeyEvent.VK_SHIFT);
+        robot.keyPress(KeyEvent.VK_CONTROL);
+        robot.keyPress(keySelected);
+        robot.keyRelease(keySelected);
+        robot.keyRelease(KeyEvent.VK_SHIFT);
+        robot.keyRelease(KeyEvent.VK_CONTROL);
+        robot.waitForIdle();
+        robot.delay(100);
+
+        SwingUtilities.invokeAndWait(() -> {
+            selectedColumnAfterKeyPress = table.getSelectedColumns();
+        });
+
+        if (selectedColumnAfterKeyPress[0] != startCellCheck ||
+                selectedColumnAfterKeyPress[selectedColumnAfterKeyPress.length - 1] !=
+                        endCellCheck) {
+            System.out.println("Selected Columns: ");
+            for (int columnsSelected : selectedColumnAfterKeyPress) {
+                System.out.println(columnsSelected);
+            }
+            String failureMsg = "Test Failure. Failed to select cells for Ctrl" +
+                    " Shift " + key + " selection";
+            throw new RuntimeException(failureMsg);
```

Comment:
Suggestion: A StringBuffer can be used to log the errors and the RuntimeException can be thrown at the end after checking both cases- left and right selection similar to this test https://github.com/openjdk/jdk/blob/47c8a6a8db979fe862be876008feb76cdc9dccfd/test/jdk/javax/swing/plaf/metal/MetalBorders/ScaledMetalBorderTest.java#L82

Reply:
Noted, StringBuffer can be used to log errors. But I don't think it's required here since we are not appending anything to Error message and also when either of selection fails its better to throw exception without proceeding to other side of test.